src/decoder/read_seek_source.rs

```diff
@@ -5,13 +5,29 @@ use symphonia::core::io::MediaSource;
 
 pub struct ReadSeekSource<T: Read + Seek + Send + Sync> {
     inner: T,
+    byte_len: Option<u64>,
+}
+
+// Copied from std Seek::stream_len since its unstable
+fn stream_len(stream: &mut impl Seek) -> std::io::Result<u64> {
+    let old_pos = stream.stream_position()?;
+    let len = stream.seek(SeekFrom::End(0))?;
+
+    // Avoid seeking a third time when we were already at the end of the
```

Comment:
Why a third time?

Reply:
This is the source: https://github.com/rust-lang/rust/blob/master/library/std/src/io/mod.rs#L1841

Third time is needed to go back to initial position

Reply:
Well, i guess its only 2 seeks, so its second, not third, bug in a comment in std I guess :)

Reply:
Actually, stream_position() is also calling a seek, so it is indeed a third

Reply:
Are you asking about `MediaSource::byte_len` method? Yes, it is provided by symphonia, and this PR changes the implementation to return `Some` calculated length now instead of returning `None` as before. This is then used to estimate the duration of the track

The method says that "This may be an expensive operation"

Although in my implementation here, I cache it when creating the source instead of calculating when the method is called.
I did it because of the method requiring shared reference to self instead of a `&mut` reference.

This could instead be done with a `RefCell` so this "expensive" operation is performed only when actually needed

Would that be better?

Other ways I see is changing the API bringing a trait like `MediaSource` into rodio itself. so that implementation of this is defined by the user, but I don't think thats a good idea

Reply:
I like the RefCell idea tbh.

Reply:
This would also make total_duration() work for all symphonia-decoded audio, which would be really helpful for my projects. @kuviman if you want me to try make the refcell changes, I'd be happy to.

Reply:
I have updated the PR to use lazy calculation. This is done not using `RefCell`, but with a `Mutex`/`RwLock` since we need the source to by `Sync`.

This should not have any performance issues tho since `get_mut` is enough and works without locking for normal usage, and locking will only happen once when we need to calculate the stream length